OCPBUGS-1849: Move section about disabling Transparent Huge Pages

jmencak opened the pull request:
Moving the section about disabling THP into scalability_and_performance and adjusting the IBM Z instructions.

Other changes:
 - lower the priority to 25 to make the configuration work on SNO clusters nodes.
 - dropping z-systems specific info.
 - removing the deletion of the config which was confusing and would roll-back the changes.

Version(s):
4.8+

Links to docs preview:
 - https://51068--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/what-huge-pages-do-and-how-they-are-consumed-by-apps.html
 -  https://51068--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ibm-z-recommended-host-practices.html

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

<!-- NOTE:
Automatic preview functionality is currently only available for some branches. For PRs that update the rendered build in any way against branches that do not create an automated preview:
  * OpenShift documentation team members (core and aligned) must include a link to a locally generated preview.
  * External contributors can request a generated preview from the OpenShift documentation team. --->

<!--- Next steps after opening your PR:

* Ask for review from the OpenShift docs team:
  - For community authors: Request a review by tagging @openshift/team-documentation in a GitHub comment.
  - For Red Hat associates:
    * For normal peer requests, add a comment that contains this text: /label peer-review-needed
    * For normal merge review requests, add a comment that contains this text: /label merge-review-needed
    * For urgent peer review requests, ping @peer-review-squad requesting a review in the #forum-docs-review channel (CoreOS Slack workspace) and provide the following information:
      * A link to the PR.
      * The size of the PR that the GitHub bot assigns (ex: XS, S, M, L, XL).
      * Details about how the PR is urgent.
    * For urgent merge requests, ping @merge-review-squad in the #forum-docs-review channel (CoreOS Slack workspace).
    * Except for changes that do not impact the meaning of the content, QE review is required before content is merged.

* IMPORTANT:
  - All documentation changes must be verified by a QE team associate before merging.
  - Squash to one commit before submitting your PR for peer review.

* For more information about verifying your content, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#verification-of-your-content

* For more information about contributing to OpenShift documentation, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/contributing.adoc

Additional resources

The OpenShift docs repo adheres to the following style guides:

- OpenShift documentation guidelines (OSDOCS)
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc
- Red Hat Supplementary Style Guide (SSG)
  https://redhat-documentation.github.io/supplementary-style-guide/
- Modular Documentation Reference Guide (Mod Docs)
  https://redhat-documentation.github.io/modular-docs/
- IBM Style Guide (ISG)
  https://www.ibm.com/docs/en/ibm-style

You can log in to the ISG by using your @redhat.com id and single sign-on (SSO) credentials. --->

jmencak commented:
/jira refresh

jmencak commented:
/label peer-review-needed

jmencak commented:
Hi @jeniferh , feel free to go through the changes if you have time.

jmencak commented:
@liqcui would you be able to provide QE review?  Thank you!

liqcui commented:
/lgtm

jeniferh commented:
Thanks looks great, I think this will be very helpful!

jmencak commented:
@sheriff-rh , mind taking a look from the doc's people perspective?  Thank you!

sheriff-rh commented on modules/configuring-transparent-huge-pages.adoc:
```diff
@@ -0,0 +1,66 @@
+
+:_content-type: PROCEDURE
+[id="disable-thp_{context}"]
+= How to configure Transparent Huge Pages
```
Additionally, I think we should change the name of the file to `modules/disabling-transparent-huge-pages.adoc`.

jmencak replied:
Makes sense, I also considered doing this.  The reason I did not, is that the user may specify "madvise" or "always" instead of "never" and then he/she is not disabling them.  Only the example below shows how to disable them (one specific case of configuration).  Nevertheless, using your suggestion then.

sheriff-rh commented on scalability_and_performance/what-huge-pages-do-and-how-they-are-consumed-by-apps.adoc:
```diff
@@ -18,3 +18,5 @@ include::modules/consuming-huge-pages-resource-using-the-downward-api.adoc[level
 
 include::modules/configuring-huge-pages.adoc[leveloffset=+1]
+
+include::modules/configuring-transparent-huge-pages.adoc[leveloffset=+1]
```
If you do change the name of the file, be sure it matches here. Another reason why renaming is a good idea is because the module above is also called "configuring huge pages" and that could be confusing too.

sheriff-rh reviewed (changes requested):
Looks good - I have a few suggestions. Reach out on slack if you have any questions!

sheriff-rh commented:
/assign sheriff-rh

sheriff-rh commented:
/label peer-review-done

jmencak commented:
@sheriff-rh thank you for the review, I believe I addressed all your comments.  Could you please drive this down to 4.8+?  Many thanks!

sheriff-rh commented:
Thanks @jmencak! I'll get it merged to 4.8+.

kalexand-rh commented on modules/disabling-transparent-huge-pages.adoc:
```diff
@@ -0,0 +1,66 @@
+
+Transparent Huge Pages (THP) attempt to automate most aspects of creating, managing, and using huge pages. Since THP automatically manages the huge pages, this is not always handled optimally for all types of workloads. THP can lead to performance regressions, since many applications handle huge pages on their own. Therefore, consider disabling THP. The following steps describe how to disable THP ...
+
+[id="disable-thp-with-a-nto-profile"]
```
Remove this line and the following one - procedures only have the H1 heading.

kalexand-rh commented on modules/disabling-transparent-huge-pages.adoc:
```diff
@@ -0,0 +1,66 @@
+.Procedure
+
+. Create a file with the following content and name it `thp-disable-tuned.yaml`:
+
```
Remove this line.

kalexand-rh commented on modules/disabling-transparent-huge-pages.adoc:
```diff
@@ -0,0 +1,66 @@
+
+* Log in to one of the nodes and do a regular THP check to verify if the nodes applied the profile successfully:
++
+[source,terminal]
```
You can have only one command per codeblock - please separate these. Consider creating substeps that describe the different actions.

kalexand-rh replied:
That does not look like a single command that someone could copy and paste.

If it's two commands, it should look like this:

```
[source,terminal]
----
$ cat /sys/kernel/mm/transparent_hugepage/enabled
----
+
[source,terminal]
----
$ always madvise [never]
----
```

If it's a command and it's output, it should look like this:
```
[source,terminal]
----
$ cat /sys/kernel/mm/transparent_hugepage/enabled
----
+
.Example output
[source,terminal]
----
always madvise [never]
----
```

jmencak replied:
Thank you, it's a command and its output.  Changed.

kalexand-rh commented on modules/disabling-transparent-huge-pages.adoc:
```diff
@@ -0,0 +1,66 @@
+    profile: openshift-thp-never-worker
+----
+
+. Create the Tuned object:
```
Is Tuned meant to be capitalized here?

jmencak replied:
Yes.

kalexand-rh commented on modules/ibm-z-disable-thp.adoc:
```diff
@@ -4,71 +4,6 @@
 
 :_content-type: PROCEDURE
```
This is no longer a procedure. I think it's closest to a concept.

It's also making the assembly that currently contains this module less useful. I think you should also add the new procedure to `scalability_and_performance/ibm-z-recommended-host-practices.adoc` or add an additional resources section to that assembly that contains a link to the Huge Pages assembly.

jmencak replied:
s/PROCEDURE/CONCEPT/

jmencak replied:
Added an additional resources section to that assembly that contains a link to the Huge Pages assembly.

kalexand-rh reviewed (commented):
This is close, but it needs a few more changes.

jmencak commented:
@kalexand-rh , thank you for the review, the comments were very helpful.  I believe I've addressed all of your concerns.  Hopefully there are no other outstanding issues with this PR.

jmencak commented on scalability_and_performance/what-huge-pages-do-and-how-they-are-consumed-by-apps.adoc:
```diff
@@ -18,3 +18,5 @@ include::modules/consuming-huge-pages-resource-using-the-downward-api.adoc[level
 * xref:../nodes/containers/nodes-containers-downward-api.adoc#nodes-containers-downward-api[Allowing containers to consume Downward API objects]
 
 include::modules/configuring-huge-pages.adoc[leveloffset=+1]
+
```
You lost me here, sorry.  Perhaps the docs writers can do that in a separate PR?  Or could you suggest a change as you did below?  Thank you.

kalexand-rh replied:
I can't suggest the change because it's on a line that isn't covered. If the build passes without the change, we can open a follow-up.

jmencak commented on scalability_and_performance/ibm-z-recommended-host-practices.adoc:
```diff
@@ -71,3 +71,5 @@ include::modules/ibm-z-rhel-kvm-host-recommendations.adoc[leveloffset=+1]
 
 * link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_virtualization/getting-started-with-virtualization-in-rhel-8-on-ibm-z_configuring-and-managing-virtualization[Getting started with virtualization on IBM Z]
+
+* xref:../scalability_and_performance/what-huge-pages-do-and-how-they-are-consumed-by-apps.adoc[What huge pages do and how they are consumed by applications]
```
Thank you, accepted your suggestion.

jmencak replied:
I've removed the Additional resources completely to see if the build passes.  I believe it may be the xref issues Andrew mentioned above.

jmencak commented:
@sheriff-rh , anything else I can do to move this PR forward?  Thank you!

sheriff-rh commented:
@kalexand-rh I believe Jiri has addressed all of your feedback and is ready for merge when you get a chance. Thanks!

kalexand-rh commented:
/cherrypick enterprise-4.8

kalexand-rh commented:
/cherrypick enterprise-4.9

kalexand-rh commented:
/cherrypick enterprise-4.10

kalexand-rh commented:
/cherrypick enterprise-4.12
